src/Sylius/Component/Resource/Repository/InMemoryRepository.php

```diff
@@ -0,0 +1,245 @@
+    }
+
+    /**
+     * @param ResourceInterface $resource
+     */
+    public function remove(ResourceInterface $resource)
+    {
+        if ($this->arrayObject->offsetExists($resource->getId())) {
+            $this->arrayObject->offsetUnset($resource->getId());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function find($id)
+    {
+        if (empty($id)) {
+            return null;
+        }
+
+        if ($this->arrayObject->offsetExists($id)) {
+            return $this->arrayObject->offsetGet($id);
+        }
+
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findAll()
+    {
+        return array_values($this->arrayObject->getArrayCopy());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findBy(array $criteria = array(), array $orderBy = null, $limit = null, $offset = null)
+    {
+        $results = $this->arrayObject->getArrayCopy();
+
+        if (isset($criteria)) {
+            $results = $this->applyCriteria($results, $criteria);
+        }
+
+        if (isset($orderBy)) {
+            $results = $this->applyOrder($results, $orderBy);
+        }
+
+        $results = array_slice($results, $offset, $limit);
+
+        return $results;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \UnexpectedValueException
+     */
+    public function findOneBy(array $criteria)
+    {
+        $results = $this->applyCriteria($this->findAll(), $criteria);
+
+        if (1 < $length = count($results)) {
+            throw new \UnexpectedValueException(sprintf('Non unique result. Number of results: %s.', $length));
+        }
+
+        return $results[0];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getClassName()
+    {
+        return $this->interface;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createPaginator(array $criteria = null, array $orderBy = null)
+    {
+        $results = $this->findAll();
+
+        if (isset($orderBy)) {
+            $results = $this->applyOrder($results, $orderBy);
+        }
+
+        if (isset($criteria)) {
+            $results = $this->applyCriteria($results, $criteria);
+        }
+
+        $adapter = new ArrayAdapter($results);
+        $pagerfanta = new Pagerfanta($adapter);
+
+        return $pagerfanta;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createNew()
+    {
+    }
+
+    /**
+     * @param ResourceInterface[] $resources
+     * @param array               $criteria
+     *
+     * @return ResourceInterface[]|array
+     */
+    private function applyCriteria(array $resources, array $criteria = array())
+    {
+        foreach ($this->arrayObject as $object) {
+            foreach ($criteria as $criterion => $value) {
+                if ($value !== $this->accessor->getValue($object, $criterion)) {
+                    $key = array_search($object, $resources);
+                    unset($resources[$key]);
```

Comment:
Yup, and tested the performance, but in this case `array_filter` is slower than combining `array_search` with `unset`.